File: Sales_Recon/src/pandas/0_CM_사급마감_PIR.py
import pandas as pd
from pathlib import Path
from janitor import clean_names


# Path
path = Path(__file__).parent.parent.parent


# Functions
def rename_columns(df):
    df = df.rename(
        columns={
            "unnamed_14": "pir_net_price",
            "unnamed_22": "sa_net_price",
            "unnamed_25": "unit",
            "unnamed_35": "sa_valid_from",
            "info_record_1": "pir_valid_from",
            "sa_#": "sa_number",
        }
    )
    return df


def change_data_type(df):
    # remove NA values
    df["sa_net_price"] = df["sa_net_price"].fillna(0)
    # change from float to integer
    df["sa_net_price"] = df["sa_net_price"].astype(int)
    return df


def select_columns(df, selected_columns):
    df = df[selected_columns]
    return df


def main():

    # Variables
    from common_variable import year, month

    # Filenames
    input_file = path / "data" / "VAN CM" / f"SA_{year}{month}.xls"
    output_1 = path / "data" / "VAN CM" / "SA_LS Automotive.csv"
    output_2 = path / "data" / "VAN CM" / "SA_MOBASE.csv"

    # Read data
    df = pd.read_csv(
        input_file,
        skiprows=9,
        sep="\t",
        encoding="UTF-16LE",
        skipinitialspace=True,
        thousands=",",
        engine="python",
    ).clean_names(strip_accents=False)

    # Process data
    df = (
        df.pipe(rename_columns)
        .pipe(change_data_type)
        .pipe(
            select_columns,
            [
                "vendor",
                "vendor_name",
                "material_no",
                "mtye",
                "sa_net_price",
                "unit",
                "sa_valid_from",
                "sa_number",
            ],
        )
    )

    ## Filter dataframe
    df = df[df["sa_net_price"] != 0]
    df_ls = df[df["vendor"] == "9139976"]
    df_mo = df[df["vendor"] == "9082855"]

    # Write data
    df_ls.to_csv(output_1, index=False)
    df_mo.to_csv(output_2, index=False)
    print("Files are created")


if __name__ == "__main__":
    main()

File: Sales_Recon/src/dplyr/0_CM_사급마감_PIR.R
library(dplyr)
library(readr)
library(writexl)
library(janitor)
library(here)
library(glue)


# Path
path <- here("Sales_Recon")


# Functions
rename_columns <- function(df) {
  df <- df |>
    rename(
      "pir_net_price" = "x15",
      "sa_net_price" = "x23",
      "unit" = "x26",
      "sa_valid_from" = "x36",
      "pir_valid_from" = "info_record_40",
    )
  return(df)
}


main <- function() {
  # Variables
  source(here(path, "src", "dplyr", "common_variable.R"))

  # Filenames
  input_file <- here(path, "data", "VAN CM", glue("SA_{year}{month}.xls"))
  output_1 <- here(path, "data", "VAN CM", "SA_LS Automotive.xlsx")
  output_2 <- here(path, "data", "VAN CM", "SA_MOBASE.xlsx")

  # Read data
  df <- read_tsv(input_file,
    skip = 9,
    locale = locale(encoding = "UTF-16LE"),
    show_col_types = FALSE,
  ) |> clean_names(ascii = FALSE)

  # Process data
  df <- df |> 
    rename_columns() |>
    # select columns
    select(c(
      "vendor",
      "vendor_name",
      "material_no",
      "mtye",
      "sa_net_price",
      "unit",
      "sa_valid_from",
      "sa_number"
    ))

  ## Filter dataframe
  df <- df |>
    filter(.data$sa_net_price != 0)

  df_ls <- df |>
    filter(.data$vendor == "9139976")

  df_mo <- df |>
    filter(.data$vendor == "9082855")

  # Write data
  write_xlsx(df_ls, output_1)
  write_xlsx(df_mo, output_2)
  print("Files are created")
}

main()
